src/goto-harness/memory_snapshot_harness_generator.h

```diff
@@ -69,6 +69,46 @@ class memory_snapshot_harness_generatort : public goto_harness_generatort
     override;
 
 protected:
+  /// User provided goto location: function name and (maybe) location number;
+  /// the structure wraps this option with a parser
+  struct entry_goto_locationt
+  {
+    irep_idt function_name;
+    optionalt<unsigned> location_number;
+
+    entry_goto_locationt(const std::string &cmdl_option);
```

Comment:
`explicit` - also, I'm personally not a huge fan of "do work" constructors, i.e. this one parses from string, I'd prefer if it was a function that said it did that in its name

Reply:
Done + constructors are now trivial.